arb_os/gasAccounting.mini

```diff
@@ -886,111 +907,160 @@ public impure func getNextRequestFromCongestionAuction() -> (
+        false
     };
 
-    if ( ! feeStats.noFeeGasEstimationMode) {
-        acctStore = xif let Some(store) = accountStore_transferEthBalance(
-            acctStore,
-            payer,
-            feeConfig.forNetwork,
-            payableToFee,   // safe because payerBalance >= amount and feeConfig.shareForNetFee <= 1
-        ) {
-            store
-        } else {
-            panic  // should never happen -- we just checked the balance
-        };
+    let acctStore = getGlobalAccountStore();
+    let payerAcct = accountStore_get(acctStore, payer);
+    if ((account_getEthBalance(payerAcct) < totalDue) && !dontActuallyPay){
+        totalDue = account_getEthBalance(payerAcct);
+        fullyPaid = false;
+    }
 
-        if (amount > payableToFee) {
-            // Reimburse the aggregator for its costs for this tx.
-            feeStats = feeStats with { paidAggregator: recipient };
-            acctStore = xif let Some(store) = accountStore_transferEthBalance(
-                acctStore,
-                payer,
-                accountStore_feeCollectorForAggregator(acctStore, recipient),
-                amount - payableToFee,  // safe because payableToFee <= amount <= payer's original balance
-            ) {
-                store
-            } else {
-                panic   // should never happen -- we have deducted amount (in total) <= payerBalance
-            };
-        }
+    if (! fullyPaid) {
+        // pro-rate the actual payment amount between tx and calldata
+        weiForTx = totalDue * weiForTx / (weiForTx + weiForCalldata);
+        weiForCalldata = totalDue - weiForTx;
     }
 
-    setGlobalAccountStore(acctStore);
-    let amountForTx = prices.perL2Tx * amount / fullAmount;   // safe because fullAmount > 0
-    return (
-        amount == fullAmount,
-        feeStats with {
-            units: feeStats.units with {
-                tx: 1
-            } with {
-                calldataUnit: calldataUnits
-            }
-        } with {
-            weiPaid: feeStats.weiPaid with {
-                tx: amountForTx
-            } with {
-                calldataUnit: amount - amountForTx   // safe because amount >= amountForTx
-            }
-        }
-    );
+    return xif let Some(uAcctStore) = accountStore_transferEthBalance(
+        acctStore,
+        payer,
+        recipient,
+        xif (dontActuallyPay) { 0 } else { totalDue }
+    ) {
+        setGlobalAccountStore(uAcctStore);
+        (
+            fullyPaid,
+            feeStats with {
+                weiPaid: feeStats.weiPaid with {
+                    tx: feeStats.weiPaid.tx + weiForTx
+                } with {
+                    calldataUnit: feeStats.weiPaid.calldataUnit + weiForCalldata
+                }
+            },
+            totalDue,
+        )
+    } else {
+        // this should never happen, because we reduced totalDue to fit within payer's funds
+        // so just do something safe
+        (
```

Comment:
Could we add a healthcheck or at least debugprint here?

Reply:
Added a debugprint.